### DISCUSSION TITLE
Selecting a kit with variant

### DISCUSSION THREAD

**Discussion by eMKa94** [microsoft/vscode-cmake-tools#4134] [Ideas]
I'm working mostly on embedded projects where, besides the cross-compilation config, I also have the variants for unit tests and sometimes some host console applications within a single project. I often have a variant for "application type" where I can choose if I'm building the firmware or unit test etc... Swapping between those variants also requires manually changing the selected kit and if 
only I select a new variant ( or KIT ) the configuration starts and it ends up with errors because for example I change to the firmware build variant with native compiler kit selected or change the kit to cross-compiler but the variant was set to build unit tests for host machine... In my imagination, it will be perfect if I will be able to point the correct kit when defining the variant option in cmake-variants.json

**Comment by gcampbell-msft** [comment]
@eMKa94 Overall, I would suggest that you migrate to CMakePresets.json, I think this will be easier and this is officially supported by Kitware (developers of CMake). 

Thanks! If you have any other issues or think it's a bug with the kits/variants experience, please create an issue and we'd be happy to investigate.
